test/fuzz/basic.ml

```diff
@@ -477,22 +477,22 @@ let expr_call (env : Env.t) (stack : val_type list) =
       | _ -> None )
     env.funcs
 
-let expr_br_if (env : Env.t) (stack : val_type list) (is_block : bool) =
+let expr_br_if (env : Env.t) (stack : val_type list) (bkind : Env.block_kind) =
   match stack with
   | [] -> []
   | _hd::tl ->
-    let blocks_type = if is_block then env.blocks else env.loops in
     List.filter_map
-      (fun (name, bt) ->
+      (fun (_, name, bt) ->
         match bt with
         | Arg.Bt_raw (_, (pt, rt)) when
-          if is_block then S.is_stack_compatible tl (List.rev rt)
-          else S.is_stack_compatible_param tl (List.rev pt) ->
-            Some ( pair
+          match bkind with
+          | Env.Block -> S.is_stack_compatible tl (List.rev rt)
+          | Env.Loop -> S.is_stack_compatible_param tl (List.rev pt)
```

Comment:
I don't understand how this compile. I guess you're missing a `when` instead of the `->`.

Anyway, you should add a `| Env.Func` case, that would branch on a `Raw` indice (a `Symbolic` won't work). In order to get the indice you must do something like `List.length @@ Env.get_blocks env`.

Reply:
Oh indeed, sorry. This is a little bit confusing:

```ocaml
| Env.Loop -> S.is_stack_compatible_param stack (List.rev pt)
-> ...
```


Maybe you should remove the `when` and do something like:

```ocaml
match bt with
  | Arg.Bt_raw (_, (pt, rt)) ->
    let is_stack_compatible = match bkind with
    | Env.Block -> S.is_stack_compatible stack (List.rev rt)
    | Env.Loop -> S.is_stack_compatible_param stack (List.rev pt)
    in
    if not is_stack_compatible then None
    else ...
```